tests/test_docs/test_commands.py

```diff
@@ -0,0 +1,181 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2022 Valory AG
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains the tests for the commands in the documentation."""
+
+import itertools
+import re
+from copy import deepcopy
+from pathlib import Path
+from typing import Dict, Union
+
+import click
+from click.core import Command, Group, Option
+
+from aea.cli import cli as aea_cli
+from scripts.check_doc_ipfs_hashes import read_file
+
+
+def get_cmd_data(cmd: Union[Command, Group]) -> Dict:
+    """Returns a dict containing the command options and arguments."""
+
+    return {
+        "commands": {},
+        "options": list(
+            itertools.chain(*[i.opts for i in cmd.params if isinstance(i, Option)])
+        )
+        + ["--help"],  # we add help here as it does not appear in the list
+        "arguments": list(
+            itertools.chain(
+                *[i.opts for i in cmd.params if isinstance(i, click.Argument)]
+            )
+        ),
+    }
+
+
+def get_group_tree(cmd: click.Group) -> Dict:
+    """Returns a tree containing the command data."""
+    tree = get_cmd_data(cmd)
+
+    if isinstance(cmd, click.Group):
+
+        for sub_cmd_name in cmd.list_commands(click.Context):
+
+            # Get the sub-command
+            sub_cmd = cmd.get_command(click.Context, sub_cmd_name)
+
+            # Recursively build the tree
+            tree["commands"][sub_cmd_name] = get_group_tree(sub_cmd)
+
+    return tree
+
+
+class CommandValidator:
+    """Validates commands against a CLI"""
```

Comment:
Why can we not execute the command instead?

Reply:
Because in some cases that will take too much time and will need a lot of setup.